interfaces/Portalicious/src/app/components/query-table/query-table.component.ts

```diff
@@ -484,9 +489,43 @@ export class QueryTableComponent<TData extends { id: PropertyKey }, TContext> {
    */
   visibleColumns = model<QueryTableColumn<TData>[]>([]);
 
-  resetColumnVisibility() {
+  private getStoredColumns(stateKey: string): null | QueryTableColumn<TData>[] {
+    const storedColumns = localStorage.getItem(stateKey);
+    if (!storedColumns) return null;
+
+    return JSON.parse(storedColumns) as QueryTableColumn<TData>[];
+  }
+
+  private getMatchingColumns(
+    storedColumns: QueryTableColumn<TData>[],
+  ): QueryTableColumn<TData>[] {
+    return storedColumns
+      .map((column) => this.columns().find((c) => c.field === column.field))
+      .filter(Boolean) as QueryTableColumn<TData>[];
+  }
+
+  private getDefaultColumns(): QueryTableColumn<TData>[] {
+    return this.columns().filter((column) => !column.defaultHidden);
+  }
+
+  resetColumnVisibility(revertToDefault = false): void {
```

Comment:
Yeah this function is way better indeed. I think the only thing is maybe I would rename it now
```suggestion
  updateColumnVisibility(revertToDefault = false): void {
```